shell/server/proxy.js

```diff
@@ -1620,8 +1614,96 @@ Proxy.prototype.handleRequest = function (request, data, response, retryCount) {
       }, context);
     } else if (request.method === "DELETE") {
       return session.delete(path, context);
+    } else if (request.method === "PROPFIND") {
+      return session.propfind(path, {
+        mimeType: request.headers["content-type"] || "application/octet-stream",
+        content: data,
+        encoding: request.headers["content-encoding"]
+      }, context);
+    } else if (request.method === "PROPPATCH") {
+      return session.proppatch(path, {
+        mimeType: request.headers["content-type"] || "application/octet-stream",
```

Comment:
If we're going to do this repeatedly, it'd be nice to extract `mimeType` into a local variable. Not quite sure why it's needed, though; would love to hear more about that.